<?xml version="1.0" encoding="UTF-8"?>
<launch>
    
  <!-- Arguments for spawn_in_free_space_node with defaults provided -->
  <!-- The name of the package where SDF file is located -->
  <arg name="package_name" default="particle_filter"/>
  <!-- The relative path of the SDF file inside the package, including .sdf extension -->
  <arg name="relative_path" default="/urdf/turtlebot2.urdf"/>
  <!-- Assign the name of the model in Gazebo -->
  <arg name="model_name_in_simulation" default="robot_base"/>
  <!-- Lenght of one side of a square shape model/robot footprint in meters -->
  <arg name="footprint_side" default="0.4"/>

  <node name="spawn_in_free_space_node" pkg="particle_filter" type="random_spawn_in_free_space.py" output="screen">
    <!-- Set a private parameters in the node's ~/local namespace -->
    <param name="package_name" type="string" value="$(arg package_name)"/>
    <param name="relative_path" type="string" value="$(arg relative_path)"/>
    <param name="model_name_in_simulation" type="string" value="$(arg model_name_in_simulation)"/>
    <param name="footprint_side" value="$(arg footprint_side)"/>
  </node>

</launch>
